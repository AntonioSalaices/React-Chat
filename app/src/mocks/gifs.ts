import { GifData } from 'ApiData';

export const gifsResponse: GifData = {
  type: 'gif',
  id: 'RRkpCa1JGN89cuT0PX',
  url: 'https://giphy.com/gifs/uefa-europa-league-final-2022-23-RRkpCa1JGN89cuT0PX',
  slug: 'uefa-europa-league-final-2022-23-RRkpCa1JGN89cuT0PX',
  bitly_gif_url: 'https://gph.is/g/ZkLLl8P',
  bitly_url: 'https://gph.is/g/ZkLLl8P',
  embed_url: 'https://giphy.com/embed/RRkpCa1JGN89cuT0PX',
  username: 'uefa',
  source: '',
  title: 'Football Sport GIF by UEFA',
  rating: 'g',
  content_url: '',
  source_tld: '',
  source_post_url: '',
  is_sticker: 0,
  import_datetime: '2023-05-31 22:37:38',
  trending_datetime: '2023-05-31 22:40:33',
  images: {
    original: {
      height: '288',
      width: '480',
      size: '4510196',
      url: 'https://media2.giphy.com/media/RRkpCa1JGN89cuT0PX/giphy.gif?cid=430ca046gabbeg98p8qf34whxwsx2fclutgw77bc9rx9zzxg&ep=v1_gifs_search&rid=giphy.gif&ct=g',
      mp4_size: '1139354',
      mp4: 'https://media2.giphy.com/media/RRkpCa1JGN89cuT0PX/giphy.mp4?cid=430ca046gabbeg98p8qf34whxwsx2fclutgw77bc9rx9zzxg&ep=v1_gifs_search&rid=giphy.mp4&ct=g',
      webp_size: '1951998',
      webp: 'https://media2.giphy.com/media/RRkpCa1JGN89cuT0PX/giphy.webp?cid=430ca046gabbeg98p8qf34whxwsx2fclutgw77bc9rx9zzxg&ep=v1_gifs_search&rid=giphy.webp&ct=g',
      frames: '46',
      hash: '4bfa92193eef6ea8fc0e9a32f7743341',
    },
    fixed_height: {
      height: '200',
      width: '333',
      size: '1846726',
      url: 'https://media2.giphy.com/media/RRkpCa1JGN89cuT0PX/200.gif?cid=430ca046gabbeg98p8qf34whxwsx2fclutgw77bc9rx9zzxg&ep=v1_gifs_search&rid=200.gif&ct=g',
      mp4_size: '623610',
      mp4: 'https://media2.giphy.com/media/RRkpCa1JGN89cuT0PX/200.mp4?cid=430ca046gabbeg98p8qf34whxwsx2fclutgw77bc9rx9zzxg&ep=v1_gifs_search&rid=200.mp4&ct=g',
      webp_size: '1090934',
      webp: 'https://media2.giphy.com/media/RRkpCa1JGN89cuT0PX/200.webp?cid=430ca046gabbeg98p8qf34whxwsx2fclutgw77bc9rx9zzxg&ep=v1_gifs_search&rid=200.webp&ct=g',
    },
    fixed_height_downsampled: {
      height: '200',
      width: '333',
      size: '251898',
      url: 'https://media2.giphy.com/media/RRkpCa1JGN89cuT0PX/200_d.gif?cid=430ca046gabbeg98p8qf34whxwsx2fclutgw77bc9rx9zzxg&ep=v1_gifs_search&rid=200_d.gif&ct=g',
      webp_size: '158090',
      webp: 'https://media2.giphy.com/media/RRkpCa1JGN89cuT0PX/200_d.webp?cid=430ca046gabbeg98p8qf34whxwsx2fclutgw77bc9rx9zzxg&ep=v1_gifs_search&rid=200_d.webp&ct=g',
    },
    fixed_height_small: {
      height: '100',
      width: '167',
      size: '600088',
      url: 'https://media2.giphy.com/media/RRkpCa1JGN89cuT0PX/100.gif?cid=430ca046gabbeg98p8qf34whxwsx2fclutgw77bc9rx9zzxg&ep=v1_gifs_search&rid=100.gif&ct=g',
      mp4_size: '214092',
      mp4: 'https://media2.giphy.com/media/RRkpCa1JGN89cuT0PX/100.mp4?cid=430ca046gabbeg98p8qf34whxwsx2fclutgw77bc9rx9zzxg&ep=v1_gifs_search&rid=100.mp4&ct=g',
      webp_size: '362986',
      webp: 'https://media2.giphy.com/media/RRkpCa1JGN89cuT0PX/100.webp?cid=430ca046gabbeg98p8qf34whxwsx2fclutgw77bc9rx9zzxg&ep=v1_gifs_search&rid=100.webp&ct=g',
    },
    fixed_width: {
      height: '120',
      width: '200',
      size: '793968',
      url: 'https://media2.giphy.com/media/RRkpCa1JGN89cuT0PX/200w.gif?cid=430ca046gabbeg98p8qf34whxwsx2fclutgw77bc9rx9zzxg&ep=v1_gifs_search&rid=200w.gif&ct=g',
      mp4_size: '287627',
      mp4: 'https://media2.giphy.com/media/RRkpCa1JGN89cuT0PX/200w.mp4?cid=430ca046gabbeg98p8qf34whxwsx2fclutgw77bc9rx9zzxg&ep=v1_gifs_search&rid=200w.mp4&ct=g',
      webp_size: '492250',
      webp: 'https://media2.giphy.com/media/RRkpCa1JGN89cuT0PX/200w.webp?cid=430ca046gabbeg98p8qf34whxwsx2fclutgw77bc9rx9zzxg&ep=v1_gifs_search&rid=200w.webp&ct=g',
    },
    fixed_width_downsampled: {
      height: '120',
      width: '200',
      size: '111216',
      url: 'https://media2.giphy.com/media/RRkpCa1JGN89cuT0PX/200w_d.gif?cid=430ca046gabbeg98p8qf34whxwsx2fclutgw77bc9rx9zzxg&ep=v1_gifs_search&rid=200w_d.gif&ct=g',
      webp_size: '67194',
      webp: 'https://media2.giphy.com/media/RRkpCa1JGN89cuT0PX/200w_d.webp?cid=430ca046gabbeg98p8qf34whxwsx2fclutgw77bc9rx9zzxg&ep=v1_gifs_search&rid=200w_d.webp&ct=g',
    },
    fixed_width_small: {
      height: '60',
      width: '100',
      size: '218262',
      url: 'https://media2.giphy.com/media/RRkpCa1JGN89cuT0PX/100w.gif?cid=430ca046gabbeg98p8qf34whxwsx2fclutgw77bc9rx9zzxg&ep=v1_gifs_search&rid=100w.gif&ct=g',
      mp4_size: '49172',
      mp4: 'https://media2.giphy.com/media/RRkpCa1JGN89cuT0PX/100w.mp4?cid=430ca046gabbeg98p8qf34whxwsx2fclutgw77bc9rx9zzxg&ep=v1_gifs_search&rid=100w.mp4&ct=g',
      webp_size: '153596',
      webp: 'https://media2.giphy.com/media/RRkpCa1JGN89cuT0PX/100w.webp?cid=430ca046gabbeg98p8qf34whxwsx2fclutgw77bc9rx9zzxg&ep=v1_gifs_search&rid=100w.webp&ct=g',
    },
  },
  user: {
    avatar_url: 'https://media1.giphy.com/avatars/uefa/CtlONIU0BbPt.png',
    banner_image: 'https://media1.giphy.com/headers/uefa/IdHUPkD73XM1.gif',
    banner_url: 'https://media1.giphy.com/headers/uefa/IdHUPkD73XM1.gif',
    profile_url: 'https://giphy.com/uefa/',
    username: 'uefa',
    display_name: 'UEFA',
    description: 'The official home of the UEFA competitions on GIPHY',
    instagram_url: 'https://instagram.com/uefachampionsleague',
    website_url: 'http://www.uefa.com',
    is_verified: true,
  },
  analytics_response_payload:
    'e=Z2lmX2lkPVJSa3BDYTFKR044OWN1VDBQWCZldmVudF90eXBlPUdJRl9TRUFSQ0gmY2lkPTQzMGNhMDQ2Z2FiYmVnOThwOHFmMzR3aHh3c3gyZmNsdXRndzc3YmM5cng5enp4ZyZjdD1n',
  analytics: {
    onload: {
      url: 'https://giphy-analytics.giphy.com/v2/pingback_simple?analytics_response_payload=e%3DZ2lmX2lkPVJSa3BDYTFKR044OWN1VDBQWCZldmVudF90eXBlPUdJRl9TRUFSQ0gmY2lkPTQzMGNhMDQ2Z2FiYmVnOThwOHFmMzR3aHh3c3gyZmNsdXRndzc3YmM5cng5enp4ZyZjdD1n&action_type=SEEN',
    },
    onclick: {
      url: 'https://giphy-analytics.giphy.com/v2/pingback_simple?analytics_response_payload=e%3DZ2lmX2lkPVJSa3BDYTFKR044OWN1VDBQWCZldmVudF90eXBlPUdJRl9TRUFSQ0gmY2lkPTQzMGNhMDQ2Z2FiYmVnOThwOHFmMzR3aHh3c3gyZmNsdXRndzc3YmM5cng5enp4ZyZjdD1n&action_type=CLICK',
    },
    onsent: {
      url: 'https://giphy-analytics.giphy.com/v2/pingback_simple?analytics_response_payload=e%3DZ2lmX2lkPVJSa3BDYTFKR044OWN1VDBQWCZldmVudF90eXBlPUdJRl9TRUFSQ0gmY2lkPTQzMGNhMDQ2Z2FiYmVnOThwOHFmMzR3aHh3c3gyZmNsdXRndzc3YmM5cng5enp4ZyZjdD1n&action_type=SENT',
    },
  },
};
