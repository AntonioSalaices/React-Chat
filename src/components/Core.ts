export * from './basics/Card/Card';
export * from './basics/Input/Input';
export * from './basics/Message/Message';
export * from './basics/Spinner/Spinner';
export * from './composed/Container/Container';
export * from './composed/ErrorBoundary/ErrorBoundary';
export * from './composed/ErrorPage/ErrorPage';
export * from './composed/Footer/Footer';
export * from './composed/FormField/FormField';
export * from './composed/Header/Header';
export * from './composed/Layout/Layout';
export * from './composed/List/List';
export * from './composed/NotFound/NotFound';
export * from './composed/Section/Section';
export * from './composed/Modal/Modal';
export * from './composed/LazyComponent/LazyComponent';
