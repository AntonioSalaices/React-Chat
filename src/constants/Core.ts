export * from './authConstants';
export * from './eventConstants';
export * from './htmlConstants';
export * from './icons';
export * from './languageConstans';
export * from './routesConstants';
export * from './screenConstants';
export * from './securityConstants';
export * from './Theme';
export * from './serviceConstants';
