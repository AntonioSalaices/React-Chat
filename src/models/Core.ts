export * from './Store.ts';
export * from './Pokemon.ts';
export * from './User.ts';
