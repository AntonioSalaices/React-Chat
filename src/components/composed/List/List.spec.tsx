import { gifsResponse } from 'mocks/gifs';
import { List } from '@Components/Core';
import { render } from '@testing-library/react';

describe('<List />:: Spec', () => {
  const props = {
    isLoading: false,
    data: [gifsResponse],
    renderItem: jest.fn(),
  };

  test('should render correctly', () => {
    const component = render(<List {...props} />);
    const { container } = component;

    expect(container).toMatchSnapshot();
  });

  test('should render correctly and execute memo HOC ', () => {
    const component = render(<List {...props} />);
    const { rerender } = component;

    const otherProps = {
      ...props,
      isLoading: true,
    };

    rerender(<List {...otherProps} />);
  });
});
