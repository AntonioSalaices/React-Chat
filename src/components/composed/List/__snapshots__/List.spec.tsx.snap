// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List />:: Spec should render correctly 1`] = `
<div>
  <div
    class="row gap-2 justify-center"
  />
</div>
`;
